### ISSUE TITLE
mix test --only conflicts with ExUnit.configure( exclude: ...)  

### ISSUE THREAD

**Issue by RobertDober** [elixir-lang/elixir#4087]
I have made an empty new mix project and just added
`ExUnit.configure(exclude: :wip)` to test_helper. As a consequence `test --only <whatever>` where whatever can even assume the value of `:wip` fails as follows:

```
** (FunctionClauseError) no function clause matching in Enum.do_uniq/3
    (elixir) lib/enum.ex:2564: Enum.do_uniq(:wip, #HashSet<[:test]>, #Function<56.8839654/1 in Enum.uniq/1>)
    (elixir) lib/enum.ex:2569: Enum.do_uniq/3
    (elixir) lib/keyword.ex:558: Keyword.update/4
    (elixir) lib/keyword.ex:562: Keyword.update/4
    (mix) lib/mix/tasks/test.ex:184: Mix.Tasks.Test.run/1
    (mix) lib/mix/cli.ex:55: Mix.CLI.run_task/2
```

This is my environment

```
Erlang/OTP 18 [erts-7.1] [source] [64-bit] [smp:4:4] [async-threads:10] [kernel-poll:false]

Interactive Elixir (1.1.1) - press Ctrl+C to exit (type h() ENTER for help)
```

From the stacktrace I was guessing this being rather a mix/elixir issue than an ExUnit issue. But I am frequently wrong ;).

Sample app, for your convenience is [here](https://github.com/RobertDober/example_for_mix_test_only_issue)

**Comment by josevalim** [comment]
Thank you @RobertDober. This has been fixed in master and the currently out Elixir v1.2.0-rc version. :)

**Comment by RobertDober** [comment]
Thanx pushing me to upgrade with all those features ;)
